docs/developer-guide/working-with-plugins.md

```diff
@@ -18,6 +18,25 @@ module.exports = {
 };
 ```
 
+### Preprocessors in Plugins
+
+You can also create plugins that would tell ESLint how to process files other then JavaScript. In order to create a
+preprocessor, object that is exported from your module has to conform to the following interface:
+
+```js
+preprocessors: {
+    file_extension_without_dot: {
+        preprocess: function(string) { return [string] } //Takes text of the file and returns array of texts to be processed
+        postprocess: function([[Message]], string) { return [Message] //Takes array of array of error messages, one for
```

Comment:
Maybe add that the array items contain error messages for the corresponding array items returned from preprocess()?